src/engine/GroupBy.cpp

```diff
@@ -961,6 +971,30 @@ void GroupBy::substituteAllAggregates(
   }
 }
 
+// _____________________________________________________________________________
+template <typename A>
+concept SupportedAggregates =
+    ad_utility::SameAsAnyTypeIn<A, GroupBy::AggregationDataVectors>;
+
+template <SupportedAggregates AggregateDataVector>
+struct ResizeVectorsVisitor {
+  void operator()(AggregateDataVector& vector, size_t numberOfGroups,
+                  const GroupBy::HashMapAggregateTypeWithData& info) const {
+    (void)info;
+    vector.resize(numberOfGroups);
+  }
```

Comment:
you can get rid of the whole overloading by using a 
```
if constexpr (std::same_as<AggregateDataVector, vector<GroupConcatAggregationData>>) {
  // do the stuff that only has to be done for group concat
}
```

Then you maybe can make the `struct` a lambda , making the code shorter.